src/app/components/Checkbox/style.ts

```diff
@@ -0,0 +1,30 @@
+import tw, { css } from "twin.macro";
+
+const baseStyle = [tw`form-checkbox h-4 w-4 transition duration-150 ease-in-out`];
+
+const getColor = (color: string): any => {
```

Comment:
This whole function could probably be simplified to something like the below to avoid the repetitive `switch` statement.

```ts
const getColor = (color: string): any => {
	if (!["primary", "success", "danger", "warning"].includes(color)) {
		throw new Error(`Failed to find a color for "${color}"`);
	}

	const colorBase = `var(--theme-color-${color})`;

	return css`
		color: ${colorBase};
	`;
};
```

Reply:
Thanks for the tip!

I made the code clearer doing:

```
const getColor = (color: string = "primary"): any => {
	const colorBase = `var(--theme-color-${color})`;

	return css`
		color: ${colorBase};
	`;
};
```

Reply:
@faustbrian  If we are going to add this code snippet:

```
if (!["primary", "success", "danger", "warning"].includes(color)) {
	throw new Error(`Failed to find a color for: "${color}"`);
}
```

We should create a `<ErrorBoundary>` component.

- https://reactjs.org/docs/error-boundaries.html
- https://github.com/testing-library/react-testing-library/issues/400#issuecomment-509698450
- https://github.com/JoschuaSchneider/use-error-boundary

Reply:
@faustbrian I added this package https://github.com/bvaughn/react-error-boundary, we will use it more throughout development to create fallback errors.

I didn't add the code:

```
if (!color) {
	color = "primary";
}
```

Because there is a `defaultProps` on the component:

```
Checkbox.defaultProps = {
	type: "checkbox",
	color: "success",
};
```